addons/website/static/src/js/components/survey/survey.js

```diff
@@ -0,0 +1,807 @@
+          },
+
+          getWebsitePurpose({state}) {
+            return Object.values(state.description.descriptionData.websitePurposes).map(x => x);
+          },
+
+          getSelectedPurpose({state}, id) {
+            return id ? state.description.descriptionData.websitePurposes[id] : undefined;
+          },
+
+          getFeatures({state}) {
+            const columnNumber = 3;
+            const featureRows = [];
+
+            const features = Object.values(state.featureSelection.features);
+            let currentRow = [];
+            for (let i = 0; i < features.length; i += 1) {
+              currentRow.push(features[i]);
+              if (currentRow.length === columnNumber) {
+                featureRows.push(currentRow);
+                currentRow = [];
+              }
+            }
+            if (currentRow.length > 0) {
+              const rowLength = currentRow.length;
+              for (let i = 0; i < (columnNumber - rowLength); i += 1) {
+                currentRow.push({
+                  type: 'empty'
+                });
+              }
+              featureRows.push(currentRow);
+            }
+            return featureRows;
+          },
+
+          getPalettes({state}) {
+            const columnNumber = 4;
+            const paletteRows = [];
+            let palettes = Object.values(state.paletteSelection.palettes);
+            let currentRow = [];
+            for (let i = 0; i < palettes.length; i += 1) {
+              currentRow.push(palettes[i]);
+              if (currentRow.length === columnNumber) {
+                paletteRows.push(currentRow);
+                currentRow = [];
+              }
+            }
+            if (currentRow.length > 0) {
+              const rowLength = currentRow.length;
+              for (let i = 0; i < (columnNumber - rowLength); i += 1) {
+                currentRow.push({
+                  type: 'empty'
+                });
+              }
+              paletteRows.push(currentRow);
+            }
+            return paletteRows;
+          },
+        };
+
+        async function getInitialState(wid) {
+
+          const feautureSelection = rpc.query({
+            model: 'website.survey.feature',
+            method: 'search_read',
+            fields: ['title', 'description', 'type', 'website_type'],
+          }).then(function (results) {
+            const features = {};
+            for (let i = 0; i < results.length; i += 1) {
+              features[results[i].id] = {
+                id: results[i].id,
+                title: results[i].title,
+                description: results[i].description,
+                type: results[i].type,
+                websiteType: results[i].website_type,
+                selected: false
+              };
+            }
+            return features;
+          }).catch(function (_) {
+              return {};
+          });
```

Comment:
What kind of error are you expecting here? You should probably check that you're only ignoring the error you want and rethrow otherwise.